### PR TITLE
feat: add reason argument to pardon commands

### PR DESCRIPTION
**Pull request by VirdanTheBurden**
Closes issue #1565.

Due to the nature of the bot internals, it would be messy to only change the unban command to take a reason argument. Therefore, all pardon commands that call the method `pardon_infraction()` provided by the `InfractionScheduler` cog will have a reason argument and append the pardon reason to the infraction through `InfractionScheduler.deactivate_infraction()`'s HTTP patch request to the moderation API when the infraction is marked as non-active.

If anyone would like to see a demo of the pardon at work, here's a [video](https://youtu.be/i6iB3nTx-FQ) I've uploaded to YouTube.

The format of the pardoned text is still up for debate, as what I've done was a placeholder.

### CODE REVIEW

**Comment by VirdanTheBurden** [comment]
Oops, it seems that I've forgotten about tests - working on that now!

File: bot/exts/moderation/infraction/_scheduler.py [hunk 1]
```diff
@@ -434,9 +443,19 @@ async def deactivate_infraction(
...
+                if (punish_reason := infraction["reason"]) is not None:
+                    data["reason"] = punish_reason + " | "
+
+                data["reason"] += f"Pardoned: {pardon_reason}"
+
             await self.bot.api_client.patch(
                 f"bot/infractions/{id_}",
-                json={"active": False}
+                json=data
             )
```
Comment (wookie184) [inline bot/exts/moderation/infraction/_scheduler.py:459]: Hmm, now if `pardoned_reason` is none it'll try to update `reason` to be an empty string.

**Review by wookie184 — CHANGES_REQUESTED** [review]

File: bot/exts/moderation/infraction/_scheduler.py (hunk 1 above)
Comment (VirdanTheBurden) [reply bot/exts/moderation/infraction/_scheduler.py]: Fixed - removed the "reason" field from data so it'll just have `{"active": False}` if `pardon_reason` is `None`

Comment (wookie184) [reply bot/exts/moderation/infraction/_scheduler.py]: Now it'll error if `data["reason"] += f"Pardoned: {pardon_reason}"` is hit without reason being set first (if the reason is currently `None`) 😅 

I think setting `data["reason"] = ""` first in `if pardon_reason is not None:` would fix it.

Comment (VirdanTheBurden) [reply bot/exts/moderation/infraction/_scheduler.py]: Added.

**Review by wookie184 — APPROVED** [review]
Works nicely, thanks!

**Review by ChrisLovering — APPROVED** [review]

**Pull request merged**